agent/api/ecsclient/client.go

```diff
@@ -123,21 +123,31 @@ func (client *APIECSClient) RegisterContainerInstance(containerInstanceArn strin
 
 func (client *APIECSClient) registerContainerInstance(clusterRef string, containerInstanceArn string, attributes []string) (string, error) {
 	registerRequest := ecs.RegisterContainerInstanceInput{Cluster: &clusterRef}
+	var registrationAttributes []*ecs.Attribute
 	if containerInstanceArn != "" {
+		// We are re-connecting a previously registered instance, restored from snapshot.
 		registerRequest.ContainerInstanceArn = &containerInstanceArn
+	} else {
+		// This is a new instance, not previously registered.
+		// Custom attribute registration only happens on initial instance registration.
+		for _, attribute := range client.getCustomAttributes() {
+			log.Debug("Added a new custom attribute",
+				"name", *attribute.Name,
+				"value", *attribute.Value,
+			)
+			registrationAttributes = append(registrationAttributes, attribute)
+		}
 	}
-
+	// Standard attributes are included with all registrations.
```

Comment:
If I remember correctly, new attribute registration will overwrite existing attributes, removing ones that are no longer included.  Does this affect custom attributes too?